### PR TITLE
Process Code Async

### PR DESCRIPTION
**Pull request by thomas-huston-zocdoc**
@ZocDoc/polaris-devs 

Changes
--
- Pulled the `compileAsync` function out of `bundler.js` and made the associated tweaks to the various compile and minify tasks.

### CODE REVIEW

File: src/compile/compile-es6.js [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
  * @param {boolean} options.sourceMap
  * @param {string} options.siteRoot
- * @param {function} options.success
- * @param {function} options.error
+ * @returns {bluebird}
  */
 function compile(options) {
 
-    try {
+    return new Promise(function(resolve, reject) {
```
Comment (thomas-huston-zocdoc) [inline src/compile/compile-es6.js]: All the compile and minify tasks now return promises.

File: src/process-code/process-async.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * @param {string} options.nodeModulesPath
+ * @param {boolean} options.outputBundleOnly
+ * @param {boolean} options.outputBundleStats
+ * @param {object} options.bundleStatsCollector
+ * @param {boolean} options.sourceMap
+ * @param {string} options.siteRoot
+ * @param {boolean} options.useTemplateDirs
+ * @param {function} processFn
+ */
+function processAsync(options, processFn) {
```
Comment (thomas-huston-zocdoc) [inline src/process-code/process-async.js]: This is the old `compileAsync` function pulled out and wrapped in a promise.

File: src/bundler.js [hunk 3]
```diff
@@ -324,46 +321,57 @@ function processJsBundle(options, jsBundle, bundleDir, jsFiles, bundleName, cb)
...
-                            bundleStatsCollector.ParseJsForStats(jsBundle, jsxText);
+                            bundleStatsCollector.ParseJsForStats(jsBundle, code);
                         }
-                        getOrCreateJsx(options, jsxText, filePath, jsPathOutput, next);
-                    });
-                } else if (isES6) {
-                    jsPath = jsPathOutput;
-                    readTextFile(filePath, function(es6Text) {
+
+                        compile.jsx(compileOptions).then(next).catch(handleError);
```
Comment (Andrew-Hagedorn-ZocDoc) [inline src/bundler.js]: Is way less complicated now...

**Comment by Andrew-Hagedorn-ZocDoc** [comment]
:dart:

**Pull request merged**